HLTrigger/HLTfilters/plugins/HLTDoubletDZ.cc

```diff
@@ -347,6 +388,84 @@ bool HLTDoubletDZ<T1, T2>::getCollections(edm::Event const& iEvent,
   return false;
 }
 
+template <>
+bool HLTDoubletDZ<l1t::P2GTCandidate, l1t::P2GTCandidate>::getCollections(
+    edm::Event const& iEvent,
+    std::vector<l1t::P2GTCandidateRef>& coll1,
+    std::vector<l1t::P2GTCandidateRef>& coll2,
+    trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  const std::vector<l1t::P2GTAlgoBlock>& algos = iEvent.get(algoBlockToken_);
+
+  if (!algos.empty()) {
+    for (const l1t::P2GTAlgoBlock& algo : algos) {
```

Comment:
Not a review comment just a heads up, we are planning to move the `l1t::P2GTAlgoBlock` container from an `std::vector` to an `std::map`  which should make accessing a fair bit cleaner.